yarn-project/archiver/src/archiver/data_retrieval.ts

```diff
@@ -139,11 +140,53 @@ async function getBlockFromRollupTx(
   if (!allowedMethods.includes(functionName)) {
     throw new Error(`Unexpected method called ${functionName}`);
   }
-  const [headerHex, archiveRootHex, , , , bodyHex] = args! as readonly [Hex, Hex, Hex, Hex[], ViemSignature[], Hex];
+  // TODO(#9101): 'bodyHex' will be removed from below
+  const [headerHex, archiveRootHex, , , , bodyHex, blobInputs] = args! as readonly [
+    Hex,
+    Hex,
+    Hex,
+    Hex[],
+    ViemSignature[],
+    Hex,
+    Hex,
+  ];
 
   const header = Header.fromBuffer(Buffer.from(hexToBytes(headerHex)));
+  // TODO(#9101): Retreiving the block body from calldata is a temporary soln before we have
+  // either a beacon chain client or link to some blob store. Web2 is ok because we will
+  // verify the block body vs the blob as below.
   const blockBody = Body.fromBuffer(Buffer.from(hexToBytes(bodyHex)));
 
+  const blockFields = blockBody.toFields();
+  // TODO(#9101): The below reconstruction is currently redundant, but once we extract blobs will be the way to construct blocks.
+  // The blob source will give us blockFields, and we must construct the body from them:
+  // TODO(#8954): When logs are refactored into fields, we won't need to inject them here.
+  const reconstructedBlock = Body.fromFields(
+    blockFields,
+    blockBody.noteEncryptedLogs,
+    blockBody.encryptedLogs,
+    blockBody.unencryptedLogs,
+  );
+
+  blockBody.squashEmptyLogs();
+  if (!reconstructedBlock.toBuffer().equals(blockBody.toBuffer())) {
+    // Sometimes the below will fail because empty logs can look like:
+    //   UnencryptedTxL2Logs { functionLogs: [ UnencryptedFunctionL2Logs { logs: [] } ]} OR UnencryptedTxL2Logs { functionLogs: []}
+    // Even though they represent the same data, hence squashing the blockBody logs above
+    // TODO(#9101): Remove below check (without calldata there will be nothing to check against)
+    throw new Error(`Block reconstructed from blob fields does not match`);
```

Comment:
Atm it's checking whether the body from calldata matches the body from fields - I kept it to help finishing issue 9101 so it's easy to see where it fails if anything goes wrong. Right now its redundant but when we get block fields from the blob source rather than `block.toFields` it will have some use

Reply:
When we get block fields from blob source, the calldata will be gone and this will be removed as well? And if the fields from blob source is wrong somehow, shouldn't it fail the check below?

Reply:
Yes the check will be removed once the blob source task is complete, I just left it so it's easier to see if the blob source fails while it's being worked on.
The below would probably fail, but might pass in the case that the blobFields->block encoding is wrong but the code in foundation to convert blobFields->commitment->evaluation inputs is right.
I'm happy to remove the above check or comment it out, no problem!